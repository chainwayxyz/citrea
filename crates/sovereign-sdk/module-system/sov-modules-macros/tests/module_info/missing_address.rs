use sov_modules_api::{ModuleInfo, StateMap};

#[derive(ModuleInfo)]
struct TestStruct<C: sov_modules_api::Context> {
    #[state]
    test_state1: StateMap<u32, u32>,

    #[state]
    test_state2: StateMap<Vec<u8>, u64>,

    #[state]
    c: C,
}

fn main() {}
