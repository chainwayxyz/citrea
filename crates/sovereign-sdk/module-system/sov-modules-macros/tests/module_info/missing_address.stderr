error: The `ModuleInfo` macro requires `[address]` attribute.
 --> tests/module_info/missing_address.rs:3:10
  |
3 | #[derive(ModuleInfo)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the derive macro `ModuleInfo` (in Nightly builds, run with -Z macro-backtrace for more info)
